### PR TITLE
Requests api

### PR DESCRIPTION
**Pull request by markd315**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
New feature domain
Officers listed in Chapters

* **What is the current behavior?** (You can also link to an open issue here)
No way to request creation of auth protected resources once we turn auth on.

* **What is the new behavior (if this is a feature change)?**
Admin approval queues for creation of chapters, chapter memberships, and chapter officerships. Persistent temporary resources are either denied or promoted by authorized users.

* **Other information**:
There's no auth yet because it's waiting on Ben changes. Also this needs exploratory testing, automated tests and docs update so it's a draft for now

### CODE REVIEW

File: src/domain/requests/requests.validator.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  }
+
+  validateOfficerRequest(officerRequest) {
+    let errors = [];
+    return {
+      errors: errors,
+      isValid: (errors.length == 0)
+    }
+  }
+
```
Comment (alindenberg) [inline src/domain/requests/requests.validator.ts]: No validation on these?

Comment (markd315) [reply src/domain/requests/requests.validator.ts]: The thinking is that an advisor or admin will review them anyway before they become real objects. Maybe one for objects length >= 1 is in order anyway. Any others?

Comment (alindenberg) [reply src/domain/requests/requests.validator.ts]: If that's the case then no pressing need for anything I would say, didn't know that was the logic behind it though. Makes sense.

File: src/domain/requests/models/chapterRequest.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import Member from '../../chapter/models/chapter'
+export default interface Chapter {
```
[the remark this answers is not in the document]
Comment (markd315) [reply src/domain/requests/models/chapterRequest.ts]: Yeah, will do

[the remark this answers is not in the document]
Comment (markd315) [reply src/domain/requests/models/chapterRequest.ts]: Done

**Pull request merged**